web/modules/custom/asset_release/src/Controller/AssetReleaseController.php

```diff
@@ -73,10 +99,7 @@ public function add(RdfInterface $rdf_entity) {
    *   The access result object.
    */
   public function createAssetReleaseAccess(RdfInterface $rdf_entity) {
-    $user = $this->currentUser();
-    $membership = Og::getMembership($user, $rdf_entity);
-    // @todo: Remove check for empty after ISAICP-2369 is in.
-    return (!empty($membership) && $membership->hasPermission('create asset_release rdf_entity')) ? AccessResult::allowed() : AccessResult::forbidden();
+    return $this->ogAccess->userAccessEntity('create', $this->createNewAssetRelease($rdf_entity), $this->currentUser());
```

Comment:
@brummbar the reason this is done like this in core is because they have no other way. The permissions are not machine readable but only human readable. You cannot look at a permission and know what kind of operation is done, and on which entity. For example if you give a permission like "edit news content" to core, it has no way to figure out that the operation is "update" and the entity type is "node".

This is solved in OG, we have actual permission classes that contain all the metadata you need (see the `GroupContentOperationPermission` class). So we don't need to revert to the inferior way that this is done in core, by hard coding a human readable string like "create news content".

But I did a great effort in ensuring that it retains compatibility with the human readable permission strings in core, so your suggestion would actually work but it is not how it is intended. The reason behind the backwards compatibility is mainly for DX reasons (people are familiar with the permission strings) and to make migration from D7 to D8 easier. Internally OG will never use `OgAccess::userAccess()` for group content CRUD operations, it will always use `OgAccess::userAccessGroupContentEntityOperations()` because it is specifically designed for this use case. Create is also a CRUD operation.